src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.cpp

```diff
@@ -111,6 +111,18 @@ bool hasAllScalarValues(ArrayRef<Value> values) {
   return true;
 }
 
+/// Check if the value is a KrnlGlobalOp with a dense attribute of positive
+/// constants.
+bool indicesArePositiveConstants(Value indices) {
+  DenseElementsAttr valueAttribute =
+      krnl::getDenseElementAttributeFromKrnlValue(indices);
+  if (!valueAttribute)
+    return false;
+
+  return llvm::all_of(valueAttribute.getValues<IntegerAttr>(),
```

Comment:
It seems you assume that the elements of DenseElementsAttr are IntegerAttr. Prefer to mention that in the function's comment or explicitly handle it.

Reply:
Done.